### PR TITLE
Fix/91 bis

### LINKED ISSUE #91: Query based on index file rather than limited list of resources
Index file is an RDF file in a pod, example:
```
<http://example.com/index_xyz>
  <http://www.w3.org/2000/01/rdf-schema#seeAlso>
    <https://css5.onto-deside.ilabt.imec.be/texon/ceon/manuf1_1>,
    <https://css5.onto-deside.ilabt.imec.be/texon/ceon/manuf1_2>,
    <https://css5.onto-deside.ilabt.imec.be/texon/ceon/manuf1_3> .
```

where:
- `<http://example.com/index_xyz>` is a (configurable) subject in the index file
- `<http://www.w3.org/2000/01/rdf-schema#seeAlso>` is a (configurable) predicate
- `<https://css5.onto-deside.ilabt.imec.be/texon/ceon/manuf1_1>` (and others) are the resources to be queried by the final query

### PR DESCRIPTION
**Pull request by EmilioTR**
closes #91

### CODE REVIEW

File: src/dataProvider/SparqlDataProvider.js [hunk 1]
```diff
@@ -32,22 +79,42 @@ export default {
...
     if (meta && meta.variables) {
-      query.variableValues = meta.variables
+      query.variableValues = meta.variables;
+    }
+
+    let results;
+    try {
+      results = await executeQuery(query);
+    } catch (error) {
+      console.error("Error executing query:", error);
```
Comment (mvanbrab) [inline src/dataProvider/SparqlDataProvider.js:102]: This code change is not useful. I guess it's a remainder of a debugging session. Avoid writing to the console in finished code.

File: src/dataProvider/SparqlDataProvider.js [hunk 2]
```diff
@@ -32,22 +79,42 @@ export default {
...
+      throw error;
     }
 
-    let results = await executeQuery(query);
     if (Object.keys(filter).length > 0) {
       results = results.filter((result) => {
-        return Object.keys(filter).every((key) => {
-          return result[key] === filter[key];
-        });
+        return Object.keys(filter).every((key) => result[key] === filter[key]);
```
Comment (mvanbrab) [inline src/dataProvider/SparqlDataProvider.js:108]: Same here: although your code looks nicer, never change working code without a good reason.

File: src/dataProvider/SparqlDataProvider.js [hunk 3]
```diff
@@ -32,22 +79,42 @@ export default {
...
       });
     }
+
     const totalItems = await query.totalItems;
+
+
     return {
       data: results,
-      total: parseInt(totalItems),
+      total: parseInt(totalItems, 10),
```
Comment (mvanbrab) [inline src/dataProvider/SparqlDataProvider.js:117]: Same here; base-10 is the default anyway...

**Review by mvanbrab — CHANGES_REQUESTED** [review]
Nice solution.
My comments are minor, but I'd like you to resolve them anyway.
The failing log-in.cy.js test will pass once you merged the main branch (I fixed it there).
We'll set this branch on hold until you solved #102.

**Comment by EmilioTR** [comment]
Only the CHANGELOG.md has to be updated, appart from that everything is all right.

File: src/dataProvider/SparqlDataProvider.js [hunk 4]
```diff
@@ -32,19 +32,33 @@ export default {
...
+      }
+    }
+
     if (meta && meta.variables) {
-      query.variableValues = meta.variables
+      query.variableValues = meta.variables;
     }
 
-    let results = await executeQuery(query);
+    const results = await executeQuery(query);
```
Comment (mvanbrab) [inline src/dataProvider/SparqlDataProvider.js:50]: This should be let, not const (see reassignment in line 53)

File: src/dataProvider/SparqlDataProvider.js [hunk 5]
```diff
@@ -32,19 +32,33 @@ export default {
     query.offset = (pagination.page - 1) * pagination.perPage;
     query.sort = sort;
 
+    handleComunicaContextCreation(query);
+
+    if (query.sourcesIndex) {
+      try {
+        const additionalSources = await addComunicaContextSourcesFromSourcesIndex(query.sourcesIndex);
+        query.comunicaContext.sources = [...new Set([...query.comunicaContext.sources, ...additionalSources])];
+      } catch (error) {
```
Comment (mvanbrab) [inline src/dataProvider/SparqlDataProvider.js:41]: This doesn't make sense anymore. Remove the catch.

File: src/dataProvider/SparqlDataProvider.js [hunk 6]
```diff
@@ -365,3 +379,50 @@ async function configureBindingStream(bindingStream, variables) {
...
+        const source = binding.get('object').value;
+        if (!sourcesList.includes(source)) {
+          sourcesList.push(source);
+        }
+      });
+      bindingsStream.on('end', resolve);
+      bindingsStream.on('error', reject);
+    });
+  } catch (error) {
+    console.error("Error adding sources from index:", error);
```
Comment (mvanbrab) [inline src/dataProvider/SparqlDataProvider.js:404]: Remove the console.error call. Compose a new Error from your text here and the underlying error message text.

**Review by mvanbrab — APPROVED** [review]

**Pull request merged**